### PR TITLE
Create build pipelines for Rafter

### PR DESCRIPTION
**Pull request by m00g3n**
Create build pipelines for Rafter

Changes proposed in this pull request:

- build pipelines for Rafter

Do not merge before: https://github.com/kyma-project/rafter/pull/15/files

see also: https://github.com/kyma-project/rafter/issues/3

### CODE REVIEW

File: prow/jobs/rafter/rafter.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  kyma-project/rafter:
+    - name: post-master-rafter
+      branches:
+        - ^master$
+      <<: *job_template
+      labels:
+        <<: *job_labels_template
+        preset-build-master: "true"
+        preset-bot-github-token: "true"
+        prow.kyma-project.io/slack.skipReport: "true" # this job will be ignored by Slack reporter
```
Comment (aerfio) [inline prow/jobs/rafter/rafter.yaml]: Ignored, as in will not show in ci-force slack channel? Is this temporary, till we reach stability?

File: prow/jobs/rafter/rafter.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+postsubmits:
+  kyma-project/rafter:
+    - name: post-master-rafter
+      branches:
+        - ^master$
+      <<: *job_template
+      labels:
+        <<: *job_labels_template
+        preset-build-master: "true"
+        preset-bot-github-token: "true"
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]: Is this needed?

File: prow/jobs/rafter/rafter.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        prow.kyma-project.io/slack.skipReport: "true" # this job will be ignored by Slack reporter
+    - name: post-release-rafter
+      # Prow resolves git tags as branches, so that regexp triggers build after tagging the repository.
+      branches:
+        - 'v\d+\.\d+\.\d+$'
+      labels:
+        <<: *job_labels_template
+        preset-build-release: "true"
+        preset-bot-github-token: "true"
+        preset-kind-volume-mounts: "true"
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]: Is this needed?

File: prow/jobs/rafter/rafter.yaml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        containers:
+          - image: eu.gcr.io/kyma-project/test-infra/buildpack-golang-kubebuilder:v20190208-813daef
+            securityContext:
+              privileged: true
+            command:
+              - "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/build.sh"
+            args:
+              - "/home/prow/go/src/github.com/kyma-project/rafter"
+            env:
+              - name: RUN_ON_PROW
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]: What is the purpose of this env?

File: prow/jobs/rafter/rafter.yaml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            cpu: 0.8
+
+job_labels_template: &job_labels_template
+  preset-dind-enabled: "true"
+  preset-sa-gcr-push: "true"
+  preset-docker-push-repository-kyma: "true"
+
+presubmits: # runs on PRs
+  kyma-project/rafter:
+    - name: pre-master-rafter
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]:
```suggestion
    - name: pre-rafter
```

I don't see a point in explicitly saying that this is only of master

File: prow/jobs/rafter/rafter.yaml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      always_run: true
+      branches:
+        - ^master$
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]: All branches, not only master

File: prow/jobs/rafter/rafter.yaml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      <<: *job_template
+      labels:
+        <<: *job_labels_template
+        preset-build-pr: "true"
+
+postsubmits:
+  kyma-project/rafter:
+    - name: post-master-rafter
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]:
```suggestion
    - name: post-rafter
```

File: prow/jobs/rafter/rafter.yaml [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+      branches:
+        - ^master$
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]: All branches

File: prow/jobs/rafter/rafter.yaml [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+      <<: *job_template
+      labels:
+        <<: *job_labels_template
+        preset-build-master: "true"
+        preset-bot-github-token: "true"
+        prow.kyma-project.io/slack.skipReport: "true" # this job will be ignored by Slack reporter
+    - name: post-release-rafter
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]: Maybe: 
```suggestion
    - name: release-rafter
```

File: prow/jobs/rafter/rafter.yaml [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+      # Prow resolves git tags as branches, so that regexp triggers build after tagging the repository.
+      branches:
+        - 'v\d+\.\d+\.\d+$'
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]: It should also match suffixes

```suggestion
        - 'v\d+\.\d+\.\d+(?:-.*)?$'
```

File: prow/jobs/rafter/rafter.yaml (hunk 8 above)
Comment (m00g3n) [reply prow/jobs/rafter/rafter.yaml]: As agreed we will keep it just for master `master`

File: prow/jobs/rafter/rafter.yaml [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+job_labels_template: &job_labels_template
+  preset-dind-enabled: "true"
+  preset-sa-gcr-push: "true"
+  preset-docker-push-repository-kyma: "true"
+
+presubmits: # runs on PRs
+  kyma-project/rafter:
+    - name: pre-rafter
+      always_run: true
```
Comment (michal-hudy) [inline prow/jobs/rafter/rafter.yaml]: Add to job template, because it is also needed in post jobs

**Pull request merged**